xonsh/ptk_shell/formatter.py

```diff
@@ -5,12 +5,22 @@
 from xonsh.prompt.base import PromptFormatter, DEFAULT_PROMPT
 
 
-class ThreadedPromptFormatter(PromptFormatter):
+class PTKPromptFormatter(PromptFormatter):
```

Comment:
This is quite a bit clearer but I still think we may benefit from splitting these into two distinct classes: `AsyncPTKPromptFormatter` and `SyncPTKPromptFormatter` and remove a bunch of branching in the methods.

I guess the current way enables changing `$ENABLE_ASYNC_PROMPT` on the fly without a restart but not sure how common that would be, especially given the increased complexity here and the overhad of a bunch of if checks below per `prompt_formatter` call.

Reply:
for experimenting some try to set the variables within the prompt. Like @anki-code did

Reply:
I understand your point. I was concious to make the if checks to terminate on self.threaded (first check) and there are two places where it checks. 
regarding the additional methods, it is unavoidable, maybe we can move the methods out of classes.

Reply:
any updates ?

Reply:
I have split the code into a separate module. Now it will be more clear I think.